application/helpers/frontend_helper.php

```diff
@@ -1865,6 +1865,10 @@ function doAssessment($surveyid, $returndataonly=false)
             $groups=array();
             foreach($fieldmap as $field)
             {
+                // Init Assessment Value
+                $assessmentValue = NULL;
```

Comment:
Maybe do this only if assesment are enabled for the survey. Less code produced for a lot of page with potentially a lot of question.